src/components/ContentHolder/ContentHolderComponent.js

```diff
@@ -1,71 +1,67 @@
-import React, { useContext, useState, useCallback, useMemo } from 'react';
+import React, {  useState, useCallback } from 'react';
 import TabComponent from '../TabComponent/TabComponent';
 import MovieListComponent from '../MovieList/MovieListComponent';
 import { ErrorBoundary } from '../ErrorBoundary/ErrorBoundary';
-import { Context } from '../../contextProvider';
+import { HeaderComponent } from '../HeaderComponent/HeaderComponent';
+import MovieDetails from '../MovieDetails/MovieDetailsComponent';
 import { TABS, DEFAULT_SORT_BY, DEFAULT_FILTER_BY } from '../../consts/constants';
+import { store, getMoviesActionAsync } from '../../store/store';
+import { connect } from 'react-redux';
 import './ContentHolder.css';
 
 const ContentHolderComponent = props => {
-  const { movies } = useContext(Context);
   const [genre, setGenre] = useState(DEFAULT_FILTER_BY);
-  const [sortOrder, setSortOrder] = useState(DEFAULT_SORT_BY);
-
-  const editMovie = movie => {
-    props.setAllMovies([...props.allMovies.filter(item => item.id !== movie.id), movie]);
-  };
-
-  const deleteMovie = movie => {
-    props.setAllMovies(props.allMovies.filter(item => item.id !== movie.id));
-  };
+  const [sortBy, setSortBy] = useState(DEFAULT_SORT_BY);
+  const [movieDetails, setMovieDetails] = useState(null);
+  const [query, setQuery] = useState('');
 
   const onTabChange = index => {
     setGenre(TABS[index]);
   };
 
+  const fetchMovies = () => {
+    const filterBy = genre !== DEFAULT_FILTER_BY
+      ? [genre]
+      : null;
+    store.dispatch(getMoviesActionAsync({search:query, searchBy:'title', sortOrder:'desc', sortBy,
+      filter:filterBy}));
+  };
+
   const genreChange = useCallback(() => {
-    const moviesByGenre = genre === DEFAULT_FILTER_BY
-      ? movies
-      : movies.filter(movie => movie.genre === genre);
-    return moviesByGenre;
-  }, [genre, sortOrder, movies]);
+   fetchMovies();
+  }, [genre, sortBy, query]);
 
   const handleSort = event => {
     const sortedBy = event.target.value === DEFAULT_SORT_BY
       ? DEFAULT_SORT_BY
       : 'title';
-    setSortOrder(sortedBy);
+      setSortBy(sortedBy);
   };
+genreChange();
```

Comment:
why did you call here?? each time that it will call all items of component will be rerenders, no need to do it, we can avoid it!! 

because we try to avoid this by adding HOOKS!))